src/Rings/hilbert.jl

```diff
@@ -927,6 +927,25 @@ function separate_simple_pps(gens::Vector{PP})
 end
 
 
+# Returns nothing; throws if ker(W) contains a non-zero vector >= 0
+function _hilbert_series_check_weights(W::Vector{Vector{Int}})
+  # assumes W is rectangular (and at least 1x1)
+  # Transpose while converting:
+  ncols = length(W);
+  nrows = length(W[1]);
+  A = zero_matrix(FlintZZ, nrows,ncols);
+  for i in 1:nrows  for j in 1:ncols  A[i,j] = W[j][i];  end; end;
+  b = zero_matrix(FlintZZ, nrows,1);
+  try
+    solve_non_negative(A, b); # any non-zero soln gives rise to infinitely many, which triggers an exception
```

Comment:
So `b` is a zero vector here. But then isn't the existence of a non-trivial solution simply determined by the rank of `A`? In that case you also don't need to transpose, I guess...